### ISSUE TITLE
WNN Clustering resulting in hundreds of clusters, clusters not ordered by size 

### ISSUE THREAD

**Issue by majohn40** [satijalab/seurat#7937]
TLDR; Why is cluster 344 larger than cluster 6? I thought they were assigned cluster numbers based on cell count.

I am working with a very large (180k+) scRNA+scATAC multiome dataset. After clustering the data in WNN space using `FindClusters(mydata, graph.name = "wsnn", algorithm = 3, resolution=0.01)`, I am getting hundreds of clusters as described previously (https://github.com/satijalab/seurat/discussions/5427). The majority of these clusters have on average around 5 cells. However, cluster 344 has almost 7k cells. The count of cells per cluster is shown below.

<div class="dataTables_scrollHead" style="background-color: rgb(255, 255, 255); background-image: none !important; background-position: initial !important; background-size: initial !important; background-repeat: initial !important; background-attachment: initial !important; background-origin: initial !important; background-clip: initial !important; color: rgb(0, 0, 0); font-family: &quot;DejaVu Sans&quot;, &quot;Lucida Grande&quot;, &quot;Segoe UI&quot;, Verdana, Helvetica, sans-serif; font-size: medium; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; overflow: hidden; position: relative; border: 0px; width: 1215px;"><div class="dataTables_scrollHeadInner" style="box-sizing: content-box; width: 1215px; padding-right: 0px;">

Cluster | Number of Cells
--|--
0 | 87039
1 | 35706
2 | 18318
3 | 8267
4 | 8000
5 | 7058
344 | 6962
6 | 5155
7 | 1540
8 | 344
9 | 202

</div>

I was under the impression that the cluster numbers were assigned by the number of cells assigned to each cluster, where 0 is the cluster with the greatest number of cells- is that incorrect? Additionally, I was using algorithm 3 based on the WNN scRNA+scATAC vignette (https://satijalab.org/seurat/articles/weighted_nearest_neighbor_analysis#wnn-analysis-of-10x-multiome-rna-atac). Was there a reason that algorithm 3 was selected for this analysis over the other options?

**Comment by igrabski** [comment]
Hi Megan, you are correct that cluster numbers are assigned in order by cluster size. However, after initial cluster assignments, any singletons that are present are then re-assigned to other larger clusters. I would guess that in this case, a very large number of singletons were assigned to cluster 344, and as a result, it ended up being one of the larger clusters despite starting off much smaller.
